### PR TITLE
Refactored construction of ModuleDecl::ImportFilter.

### PR DESCRIPTION
**Pull request by jazgarewal**
Addresses [https://bugs.swift.org/browse/SR-13245?focusedCommentId=58114&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-58114](url)

### CODE REVIEW

**Comment by jazgarewal** [comment]
@varungandhi-apple Ready for you to review!

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -8275,9 +8275,10 @@ bool SwiftASTContext::CacheUserImports(SwiftASTContext &swift_ast_context,
   llvm::SmallString<1> m_description;
   llvm::SmallVector<swift::ModuleDecl::ImportedModule, 2> parsed_imports;
 
-  swift::ModuleDecl::ImportFilter import_filter;
-  import_filter |= swift::ModuleDecl::ImportFilterKind::Public;
-  import_filter |= swift::ModuleDecl::ImportFilterKind::Private;
+  swift::ModuleDecl::ImportFilter import_filter = {
+    swift::ModuleDecl::ImportFilterKind::Public,
+    swift::ModuleDecl::ImportFilterKind::Private
+  }
```
Comment (varungandhi-apple) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: Missing semicolon?

Comment (jazgarewal) [reply lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: @varungandhi-apple Fixed! Sorry about that.

**Comment by varungandhi-apple** [comment]
Could you please squash the commits? For the commit message, having something like:
```
[lldb] [NFC] Use OptionSet's new list initializer.

Addresses LLDB changes for SR-13245.
```
would arguably be more informative. (NFC = No Functional Change.)

**Comment by jazgarewal** [comment]
@varungandhi-apple All set!

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
@swift-ci please test

**Comment by varungandhi-apple** [comment]
Thanks, done!

**Comment by jazgarewal** [comment]
Awesome - thank you!

**Pull request merged**